src/components/WalletStatementModal/index.tsx

```diff
@@ -19,19 +22,42 @@ function WalletStatementModal({statementPageURL, session}: WalletStatementProps)
      * Handles in-app navigation for iframe links
      */
     const navigate = (event: MessageEvent<WalletStatementMessage>) => {
-        if (!event.data?.type || (event.data.type !== CONST.WALLET.WEB_MESSAGE_TYPE.STATEMENT && event.data.type !== CONST.WALLET.WEB_MESSAGE_TYPE.CONCIERGE)) {
+        const {data} = event;
+        const {type, url} = data || {};
+        if (!type || (type !== CONST.WALLET.WEB_MESSAGE_TYPE.STATEMENT && type !== CONST.WALLET.WEB_MESSAGE_TYPE.CONCIERGE)) {
             return;
         }
 
-        if (event.data.type === CONST.WALLET.WEB_MESSAGE_TYPE.CONCIERGE) {
-            Report.navigateToConciergeChat(true);
+        if (type === CONST.WALLET.WEB_MESSAGE_TYPE.CONCIERGE) {
+            navigateToConciergeChat(true);
         }
 
-        if (event.data.type === CONST.WALLET.WEB_MESSAGE_TYPE.STATEMENT && event.data.url) {
-            const iouRoutes = [ROUTES.IOU_REQUEST, ROUTES.IOU_SEND];
-            const navigateToIOURoute = iouRoutes.find((iouRoute) => event.data.url.includes(iouRoute));
-            if (navigateToIOURoute) {
-                Navigation.navigate(navigateToIOURoute);
+        if (type === CONST.WALLET.WEB_MESSAGE_TYPE.STATEMENT && url) {
+            const iouRoutes: Record<string, Route> = {
+                [ROUTES.IOU_REQUEST]: ROUTES.IOU_REQUEST,
+                [ROUTES.IOU_SEND]: ROUTES.IOU_SEND,
```

Comment:
Let's remove these 2 unneeded routes, also from ROUTES file bcoz it doesn't seem to be used anywhere else.